packages/jest-core/src/ViteDevServerManager.ts

```diff
@@ -0,0 +1,414 @@
+   */
+  setupHMR(): void {
+    if (!this.viteDevServer) {
+      return;
+    }
+
+    try {
+      // Listen to HMR events
+      if (this.viteDevServer.ws) {
+        this.viteDevServer.ws.on('connection', () => {
+          // Connection established
+        });
+      }
+
+      // Handle module updates
+      if (this.viteDevServer.moduleGraph) {
+        const originalOnFileChange =
+          this.viteDevServer.moduleGraph.onFileChange;
+        this.viteDevServer.moduleGraph.onFileChange = (file: string) => {
+          this.invalidateModule(file);
+          if (originalOnFileChange) {
+            originalOnFileChange.call(this.viteDevServer.moduleGraph, file);
+          }
+        };
+      }
+    } catch (error: any) {
+      console.error('Failed to setup HMR:', error.message);
+    }
+  }
+
+  /**
+   * Gets module dependencies from the module graph
+   * @param filePath - The file path to get dependencies for
+   * @returns Set of file paths that this file depends on
+   */
+  async getModuleDependencies(filePath: string): Promise<Set<string>> {
+    if (!this.viteDevServer) {
+      return new Set();
+    }
+
+    // Check cache first
+    if (this.moduleGraphCache.has(filePath)) {
+      return this.moduleGraphCache.get(filePath)!;
+    }
+
+    const dependencies = new Set<string>();
+
+    try {
+      const moduleGraph = this.viteDevServer.moduleGraph;
+      const module = moduleGraph.getModuleById(filePath);
+
+      if (module) {
+        const collectDeps = (mod: any, visited = new Set<any>()): void => {
+          if (visited.has(mod)) {
+            return;
+          }
+          visited.add(mod);
+
+          for (const imported of mod.importedModules || []) {
+            if (imported.file) {
+              dependencies.add(imported.file);
+              collectDeps(imported, visited);
+            }
+          }
+        };
+
+        collectDeps(module);
+      }
+
+      this.moduleGraphCache.set(filePath, dependencies);
+    } catch (error: any) {
+      console.error(
+        `Failed to get dependencies for ${filePath}:`,
+        error.message,
+      );
+    }
+
+    return dependencies;
+  }
+
+  /**
+   * Clears all caches (transform cache and module graph cache)
+   */
+  clearCaches(): void {
+    this.transformCache.clear();
+    this.moduleGraphCache.clear();
+  }
+
+  /**
+   * Gets the Vite module graph for dependency tracking
+   */
+  getModuleGraph(): any {
+    return this.viteDevServer?.moduleGraph;
+  }
+
+  /**
+   * Checks if the dev server is running
+   */
+  isRunning(): boolean {
+    return this.viteDevServer !== null;
+  }
+
+  /**
+   * Dynamically loads Vite module
+   */
+  private async loadVite(): Promise<any> {
+    try {
+      // Vite is an ESM module, so we need to use dynamic import
+      // This requires Jest to be configured with ESM support
+      // @ts-expect-error - Vite is an optional peer dependency
+      const viteModule = await import('vite');
```

Comment:
I wonder if we can use `https://github.com/privatenumber/tsx` to load Vite, can you investigate?

The goal is that the users don't have to run Jest in ESM mode to be able to use this experimental Vite integration